tests/test_parser.py

```diff
@@ -187,3 +190,554 @@ def test_named_color(self):
+        '1.2em bold small-caps oblique Ahem',
+        '1.2em small-caps bold oblique Ahem',
+
+        'small-caps 1.2em/3 oblique bold Ahem',
+        'small-caps 1.2em/3 bold oblique Ahem',
+        'bold 1.2em/3 small-caps oblique Ahem',
+        'bold 1.2em/3 oblique small-caps Ahem',
+        'oblique 1.2em/3 small-caps bold Ahem',
+        'oblique 1.2em/3 bold small-caps Ahem',
+
+        'small-caps 1.2em oblique bold Ahem',
+        'small-caps 1.2em bold oblique Ahem',
+        'bold 1.2em small-caps oblique Ahem',
+        'bold 1.2em oblique small-caps Ahem',
+        'oblique 1.2em small-caps bold Ahem',
+        'oblique 1.2em bold small-caps Ahem',
+
+        'small-caps bold 1.2em/3 oblique Ahem',
+        'bold small-caps  1.2em/3 oblique Ahem',
+        'oblique small-caps 1.2em/3 bold Ahem',
+        'small-caps oblique 1.2em/3 bold Ahem',
+        'oblique bold 1.2em/3 small-caps Ahem',
+        'bold oblique 1.2em/3 small-caps Ahem',
+
+        'normal normal 1.2em/3 normal Ahem',
+        'normal 1.2em/3 normal normal Ahem',
+        'normal 1.2em/3 normal Ahem',
+        '1.2em/3 normal Ahem',
+        '1.2em/3 normal normal Ahem',
+        '1.2em/3 normal normal normal Ahem',
+    ])
+
+    # Font construction test cases
+    CASE_CONSTRUCT = {
+        # <font-style> <font-variant> <font-weight> <font-size>/<line-height> <font-family>
+        'oblique small-caps bold 1.2em/3 Ahem': ('oblique', 'small-caps', 'bold', '1.2em', '3', ['Ahem']),
+        'normal small-caps bold 1.2em/3 Ahem': ('normal', 'small-caps', 'bold', '1.2em', '3', ['Ahem']),
+        'oblique normal bold 1.2em/3 Ahem': ('oblique', 'normal', 'bold', '1.2em', '3', ['Ahem']),
+        'oblique small-caps normal 1.2em/3 Ahem': ('oblique', 'small-caps', 'normal', '1.2em', '3', ['Ahem']),
+        'normal small-caps normal 1.2em/3 Ahem': ('normal', 'small-caps', 'normal', '1.2em', '3', ['Ahem']),
+        'oblique normal normal 1.2em/3 Ahem': ('oblique', 'normal', 'normal', '1.2em', '3', ['Ahem']),
+        'normal normal bold 1.2em/3 Ahem': ('normal', 'normal', 'bold', '1.2em', '3', ['Ahem']),
+        'normal normal normal 1.2em/3 Ahem': ('normal', 'normal', 'normal', '1.2em', '3', ['Ahem']),
+        'normal normal 900 1.2em/3 Ahem': ('normal', 'normal', '900', '1.2em', '3', ['Ahem']),
+    }
+    CASE_CONSTRUCT_FAMILY = {
+        'Ahem': ['Ahem'],
+        'Ahem, "White Space"': ['Ahem', 'White Space'],
+        'Ahem, "White Space", serif': ['Ahem', 'White Space', 'serif'],
+    }
+
+    @staticmethod
+    def tuple_to_font_dict(tup):
+        """Helper to convert a tuple to a font dict to check for valid outputs."""
+        font_dict = {}
+        for idx, key in enumerate(['font_style', 'font_variant', 'font_weight',
+                                   'font_size', 'line_height', 'font_family']):
+            font_dict[key] = tup[idx]
+
+        return font_dict
+
+    def test_parse_font_shorthand(self):
```

Comment:
I've been slowly coming around on pytest, in particular because it makes this kind of parameterized test much cleaner (especially in test output reporting). If you're pytest-inclined, and want to move this to a set of parametrized pytest cases, I'd be OK with that.

Reply:
Sure, I can update this